tests/cxx/iwyu_stricter_than_cpp-d3.h

```diff
@@ -0,0 +1,17 @@
+//===--- iwyu_stricter_than_cpp-d3.h - test input file for iwyu -----------===//
+//
+//                     The LLVM Compiler Infrastructure
+//
+// This file is distributed under the University of Illinois Open Source
+// License. See LICENSE.TXT for details.
+//
+//===----------------------------------------------------------------------===//
+
+#include "tests/cxx/iwyu_stricter_than_cpp-i3.h"
+#include "tests/cxx/iwyu_stricter_than_cpp-i4.h"
+
+typedef IndirectStruct3 IndirectStruct3ProvidingTypedef;
```

Comment:
>OK, I'm going to pick some more on naming. Please have patience

You are welcome!
> ... but I think the new names introduced here should be `Direct...` rather than `Indirect...`.

Do you mean typedef names? But they contain `Indirect...` just because they refer to `IndirectStruct*`.

Reply:
Yeah, it's not obvious how to best make it work. Hmm.

I think `DirectProvidingTypedefForIndirectStruct3` captures everything I'd want in the name, but it's extremely long.

I'm not overly enthusiastic about the `DoesEverythingRight` typedefs/aliases, but they do communicate their purpose with very little detail.

I've thought about this for about an hour off and on, and I can't think of anything better. Let's go with what we have.

Reply:
> Let's go with what we have.

I mean: Let's go with what's in the patch.